### DISCUSSION TITLE
Prisma Store Procedure doesn't return proper keys.

### DISCUSSION THREAD

**Discussion by lahinjs** [prisma/prisma#24261] [Q&A] [answered]
### Bug description

I created a Store Procedure with/inside mysql-client. I can see the result inside mysql-client with this command `CALL GetGroupedData()` as expected.

But when I use  const groupedData =
    await prisma.$queryRaw`CALL GetGroupedData()`;
    
I can see an array return with the keys f0, f1, f2, etc. How can I get a specific column's name as key?

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: MAC M1
- Database: MySQL 8
- Node.js version: 22

### Prisma Version

```
^5.12.1
```

**Comment by janpio** [comment]
This this covered by one of these issues?
https://github.com/prisma/prisma/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+stored+procedure+label%3A%22topic%3A+stored+procedures%22
https://github.com/prisma/prisma/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+stored+procedure+label%3A%22topic%3A+stored+procedures+in+raw%22

**Reply by lahinjs** [reply]
Yeah. But there is no solution or fix yet.

**Reply by janpio** [reply, marked as answer]
Yes, that is why these issues exist - this does not work yet. 
Please leave a 👍 reaction on the one that matches best, and maybe leave a comment with your specific use case.

**Reply by raphaelumoh** [reply]
Hi @lahinjs, you may want to take a look at this [workaround](https://github.com/prisma/prisma/issues/21570#issuecomment-1778982057).
